### PR TITLE
Fix nsys profiling in benchmark_litgpt

### PR DESCRIPTION
**Pull request by parthmannan**
<details>
</details>

## What does this PR do?

Fixes nsys profiling bug in benchmark_litgpt.py introduced in #45 
When `gradient_accumulation_steps==1`, the nsys profiler never starts the capture as that is called only if we are use higher GA steps.

### CODE REVIEW

**Review by crcrpar — CHANGES_REQUESTED** [review]

File: thunder/benchmarks/benchmark_litgpt.py [hunk 1]
```diff
@@ -387,30 +388,30 @@ def train(self):
...
+
+                if self.nsys_enabled and i == self.profiler_start and global_rank in [0, None] and step_idx == 0:
+                    print("=====Start NSYS Profiling======")
+                    torch.cuda.cudart().cudaProfilerStart()
+
+                if (step_idx + 1) % gradient_accumulation_steps != 0:
+                    with data_sync_ctx():
+                        loss = run_fwd_bwd_one_microbatch(
+                            self.model, input_ids, targets, self.gradient_accumulation_steps
+                        )
```
[the remark this answers is not in the document]
Comment (parthmannan) [reply thunder/benchmarks/benchmark_litgpt.py:404]: Ah yes, I'll make those changes! Thanks!

File: thunder/benchmarks/benchmark_litgpt.py [hunk 2]
```diff
@@ -387,16 +387,16 @@ def train(self):
             if i == self.warmup_iter:  # warmup
                 t0 = iter_t0
 
+            if self.nsys_enabled and i == self.profiler_start and global_rank in [0, None] and step_idx == 0:
```
Comment (kshitij12345) [inline thunder/benchmarks/benchmark_litgpt.py:390]: We should remove the last `step_idx == 0` as it is undefined at this point.

Comment (parthmannan) [reply thunder/benchmarks/benchmark_litgpt.py]: Ah! I got distracted with other tasks yesterday and forgot to run the code. Thanks for catching this. Fixed.

**Review by kshitij12345 — APPROVED** [review]
LGTM, thanks @parthmannan

**Comment by parthmannan** [comment]
@crcrpar @carmocca This is ready for review!

**Review by crcrpar — APPROVED** [review]

**Pull request merged**